src/components/App/index.tsx

```diff
@@ -0,0 +1,329 @@
+      filteredScheduleData,
+      setTerm,
+      setVersionName,
+      setOscar,
+      patchScheduleData
+    ]
+  );
+  const versionsContextValue = useMemo<VersionsContextValue>(
+    () => [{ ...versionLists }, { patchVersionsData }],
+    [versionLists, patchVersionsData]
+  );
+
+  // display popup when first visiting the site
+  useEffect(() => {
+    const cookieKey = 'visited-merge-notice';
+    if (!Cookies.get(cookieKey)) {
+      swal({
+        button: 'Got It!',
+        content: (
+          <div>
+            <img
+              style={{ width: '175px' }}
+              alt="GT Scheduler Logo"
+              src="/mascot.png"
+            />
+            <h1>GT Scheduler</h1>
+            <p>
+              Hey there, yellow jackets!{' '}
+              <a href="https://github.com/gt-scheduler">GT Scheduler</a> is a
+              new collaboration between{' '}
+              <a href="https://bitsofgood.org/">Bits of Good</a> and{' '}
+              <a href="https://jasonpark.me/">Jason (Jinseo) Park</a> aimed at
+              making class registration easier for everybody! Now, you can
+              access course prerequisites, instructor GPAs, live seating
+              information, and more all in one location.
+            </p>
+            <p>
+              If you enjoy our work and are interested in contributing, feel
+              free to{' '}
+              <a href="https://github.com/gt-scheduler/website/pulls">
+                open a pull request
+              </a>{' '}
+              with your improvements. Thank you and enjoy!
+            </p>
+          </div>
+        )
+      });
+
+      Cookies.set(cookieKey, 'true', { expires: 365 });
+    }
+  }, []);
+
+  // Fetch the current term's scraper information
+  useEffect(() => {
+    setOscar(null);
+    if (term) {
+      axios
+        .get(`https://gt-scheduler.github.io/crawler/${term}.json`)
+        .then((res) => {
+          const newOscar = new Oscar(res.data);
+          setOscar(newOscar);
+        });
+    }
+  }, [term]);
+
+  // Fetch all terms via the GitHub API
+  useEffect(() => {
+    axios
+      .get(
+        'https://api.github.com/repos/gt-scheduler/crawler/contents?ref=gh-pages'
+      )
+      .then((res) => {
+        const newTerms = (res.data as { name: string }[])
+          .map((content) => content.name)
+          .filter((name: string) => /\d{6}\.json/.test(name))
+          .map((name: string) => name.replace(/\.json$/, ''))
+          .sort()
+          .reverse();
+        setTerms(newTerms);
+      });
+  }, [setTerms]);
+
+  // Set the term to be the first one if it is unset
+  // (once the terms load)
+  useEffect(() => {
+    const term_from_cookie = Cookies.get('term');
+    if (!term && (!term_from_cookie || term_from_cookie === 'undefined')) {
+      const [recentTerm] = terms;
+      setTerm(recentTerm);
+    } else if (term_from_cookie && term_from_cookie !== 'undefined') {
+      setTerm(term_from_cookie);
```

Comment:
This part just seems to replicate the existing behavior of the useCookie hook. I think it's fine to leave in, though I strongly think that the code in this component is due for a re-write to have clearer loading stages (though that is out of scope of this PR).

Reply:
I agree. I only did the modification to address the problem where the initial term state is not set based on the term cookie. I believe the useCookie() hook did not set the state based on cookie before this useEffect() is called. That is why the current deployment always defaults to Fall 2021 regardless which term you last left the app at. There is probably a better way to do this, but this solution works for now.